src/assets/wise5/components/html/html-student/html-student.component.ts

```diff
@@ -34,11 +37,84 @@ export class HtmlStudent extends ComponentStudent {
     );
   }
 
-  ngOnInit() {
+  ngOnInit(): void {
     super.ngOnInit();
     this.html = this.sanitizer.bypassSecurityTrustHtml(
       this.UtilService.replaceWISELinks(this.componentContent.html)
     );
+    if (!this.isAuthoringComponentPreviewMode()) {
+      this.setupWiseLinkClickedListener();
+    }
     this.broadcastDoneRenderingComponent();
   }
+
+  setupWiseLinkClickedListener(): void {
+    // Create the wiseLinkClickedHandler and keep a reference to it so we can remove the event
+    // listener in ngOnDestroy. The removeEventListener matches the event name and callback function
+    // to remove the listener. If we use an anonymous function, it won't be able to remove the
+    // event listener.
+    this.wiseLinkClickedHandler = this.createWiseLinkClickedHandler();
+    window.addEventListener('wiselinkclicked', this.wiseLinkClickedHandler);
+  }
+
+  createWiseLinkClickedHandler(): any {
+    return (event: CustomEvent) => {
+      this.followLink(event.detail.nodeId, event.detail.componentId);
+    };
+  }
+
+  ngOnDestroy(): void {
+    if (this.currentNodeChangedSubscription != null) {
+      this.currentNodeChangedSubscription.unsubscribe();
+    }
+    if (this.wiseLinkClickedHandler != null) {
+      window.removeEventListener('wiselinkclicked', this.wiseLinkClickedHandler);
+    }
+  }
+
+  followLink(nodeId: string, componentId: string): void {
+    if (this.isLinkToComponentInStep(nodeId)) {
+      this.scrollToComponentAndHighlight(componentId);
+    } else {
+      this.goToNode(nodeId, componentId);
+    }
+  }
+
+  isLinkToComponentInStep(nodeId: string): boolean {
+    return nodeId === this.nodeId;
+  }
+
+  goToNode(nodeId: string, componentId: string): void {
+    this.currentNodeChangedSubscription = this.StudentDataService.currentNodeChanged$.subscribe(
+      () => {
+        if (componentId != '') {
+          this.scrollToComponentAndHighlight(componentId);
+        }
+      }
+    );
+    this.StudentDataService.endCurrentNodeAndSetCurrentNodeByNodeId(nodeId);
```

Comment:
Should the subscription and scroll/highlight occur outside of html-student component, like in a service? When the currentNode changes, this html-student component should no longer be alive, right? It seems strange that this subscription is alive and participating in events even after the user navigate away from the component.